students/johnpharmd/lesson08/circle_class.py

```diff
@@ -0,0 +1,54 @@
+import math
+
+
+class Circle:
+    """represents a simple circle, using properties, a class method, and
+    special methods"""
+
+    @classmethod
+    def sort(cls, radius_list):
+        return sorted(radius_list)
+
+    def __init__(self, radius):
+        self.radius = radius
+
+    def __str__(self):
+        return 'Circle with radius: ' + str(self.radius)
+
+    def __repr__(self):
+        return 'Circle(' + str(self.radius) + ')'
+
+    def __add__(self, other):
+        self.radius += other.radius
+        return self.__repr__()
+
+    def __mul__(self, value):
+        self.radius *= value
+        return self.__repr__()
+
+    def __rmul__(self, value):
+        return self.__mul__(value)
+
+    def __gt__(self, other):
+        return self.radius > other.radius
+
+    # def __lt__(self, other):
+    #     return self.radius < other.radius
+
+    def __eq__(self, other):
+        return self.radius == other.radius
+
+    @property
+    def diameter(self):
+        return self.radius * 2
+
+    @diameter.setter
+    def diameter(self, value):
+        self.radius = value / 2
+
+    @property
+    def area(self):
+        return math.pi * self.radius ** 2
+
+    def from_diameter(self):
+        return self.diameter
```

Comment:
The idea here is to let you instantiate a new circle object with only a diameter, so this should be a class method that derives the radius from the diameter and instantiates a new circle